src/prefect/cli/__init__.py

```diff
@@ -79,3 +80,22 @@ def config():
     Output your Prefect config
     """
     click.echo(prefect.config.to_dict())
+
+
+@cli.command(hidden=True)
+@click.option(
```

Comment:
This option defaults to `True`, so I'm not sure how to use it -- doesn't `prefect diagnostics` have the same behavior as `prefect diagnostics --no-secrets`? I agree the default should be to hide secrets, but that suggests to me the flag should maybe be `--include-secret-names` or something making the opt-in explicit.